backend/src/schema/resolvers/rewards.spec.js

```diff
@@ -65,74 +70,94 @@ describe('rewards', () => {
-        const variables = {
-          from: 'b6',
-          to: 'u1',
-        }
         const expected = {
-          reward: 'u1',
+          reward: {
+            id: 'u1',
+            badges: [{ id: 'indiegogo_en_rhino' }],
+          },
         }
         await expect(client.request(mutation, variables)).resolves.toEqual(expected)
       })
+
       it('rewards a second different badge to same user', async () => {
         await factory.create('Badge', {
-          id: 'b1',
-          key: 'indiegogo_en_racoon',
-          type: 'crowdfunding',
-          status: 'permanent',
+          id: 'indiegogo_en_racoon',
           icon: '/img/badges/indiegogo_en_racoon.svg',
         })
-        const variables = {
-          from: 'b1',
-          to: 'u1',
-        }
         const expected = {
-          reward: 'u1',
+          reward: {
+            id: 'u1',
+            badges: [{ id: 'indiegogo_en_racoon' }, { id: 'indiegogo_en_rhino' }],
+          },
         }
-        await expect(client.request(mutation, variables)).resolves.toEqual(expected)
+        await client.request(mutation, variables)
+        await expect(
+          client.request(mutation, {
+            ...variables,
+            from: 'indiegogo_en_racoon',
+          }),
+        ).resolves.toEqual(expected)
       })
-      it('rewards the same badge as well to another user', async () => {
-        const variables1 = {
-          from: 'b6',
-          to: 'u1',
-        }
-        await client.request(mutation, variables1)
 
-        const variables2 = {
-          from: 'b6',
-          to: 'u2',
-        }
+      it('rewards the same badge as well to another user', async () => {
         const expected = {
-          reward: 'u2',
+          reward: {
+            id: 'u2',
+            badges: [{ id: 'indiegogo_en_rhino' }],
+          },
         }
-        await expect(client.request(mutation, variables2)).resolves.toEqual(expected)
+        await expect(
+          client.request(mutation, {
+            ...variables,
+            to: 'u2',
+          }),
+        ).resolves.toEqual(expected)
       })
-      it('returns the original reward if a reward is attempted a second time', async () => {
-        const variables = {
-          from: 'b6',
-          to: 'u1',
-        }
+
+      it('creates no duplicate reward relationships', async () => {
         await client.request(mutation, variables)
         await client.request(mutation, variables)
 
-        const query = `{
-          User( id: "u1" ) {
-            badgesCount
+        const query = gql`
+          {
+            User(id: "u1") {
+              badgesCount
+              badges {
+                id
+              }
+            }
           }
-        }
         `
-        const expected = { User: [{ badgesCount: 1 }] }
+        const expected = { User: [{ badgesCount: 1, badges: [{ id: 'indiegogo_en_rhino' }] }] }
 
         await expect(client.request(query)).resolves.toEqual(expected)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-11T14:40:57Z" title="Thursday, July 11th 2019, 4:40:57 pm +02:00">Jul 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -65,74 +70,94 @@ describe('rewards', () => {
         client = new GraphQLClient(host, { headers })
       })
 
+      describe('badge for key does not exist', () => {
+        it('rejects with a telling error message', async () => {
+          await expect(
+            client.request(mutation, {
+              ...variables,
+              from: 'bullshit',
+            }),
+          ).rejects.toThrow("Couldn't find a badge with that key")
+        })
+      })
+
+      describe('user for id does not exist', () => {
+        it('rejects with a telling error message', async () => {
+          await expect(
+            client.request(mutation, {
+              ...variables,
+              to: 'bullshit',
+            }),
+          ).rejects.toThrow("Couldn't find a user with that id")
+        })
+      })
+
       it('rewards a badge to user', async () => {
-        const variables = {
-          from: 'b6',
-          to: 'u1',
-        }
         const expected = {
-          reward: 'u1',
+          reward: {
+            id: 'u1',
+            badges: [{ key: 'indiegogo_en_rhino' }],
+          },
         }
         await expect(client.request(mutation, variables)).resolves.toEqual(expected)
       })
+
       it('rewards a second different badge to same user', async () => {
         await factory.create('Badge', {
-          id: 'b1',
           key: 'indiegogo_en_racoon',
-          type: 'crowdfunding',
-          status: 'permanent',
           icon: '/img/badges/indiegogo_en_racoon.svg',
         })
-        const variables = {
-          from: 'b1',
-          to: 'u1',
-        }
         const expected = {
-          reward: 'u1',
+          reward: {
+            id: 'u1',
+            badges: [{ key: 'indiegogo_en_racoon' }, { key: 'indiegogo_en_rhino' }],
+          },
         }
-        await expect(client.request(mutation, variables)).resolves.toEqual(expected)
+        await client.request(mutation, variables)
+        await expect(
+          client.request(mutation, {
+            ...variables,
+            from: 'indiegogo_en_racoon',
+          }),
+        ).resolves.toEqual(expected)
       })
-      it('rewards the same badge as well to another user', async () => {
-        const variables1 = {
-          from: 'b6',
-          to: 'u1',
-        }
-        await client.request(mutation, variables1)
 
-        const variables2 = {
-          from: 'b6',
-          to: 'u2',
-        }
+      it('rewards the same badge as well to another user', async () => {
         const expected = {
-          reward: 'u2',
+          reward: {
+            id: 'u2',
+            badges: [{ key: 'indiegogo_en_rhino' }],
+          },
         }
-        await expect(client.request(mutation, variables2)).resolves.toEqual(expected)
+        await expect(
+          client.request(mutation, {
+            ...variables,
+            to: 'u2',
+          }),
+        ).resolves.toEqual(expected)
       })
-      it('returns the original reward if a reward is attempted a second time', async () => {
-        const variables = {
-          from: 'b6',
-          to: 'u1',
-        }
+
+      it('creates no duplicate reward relationships', async () => {
         await client.request(mutation, variables)
         await client.request(mutation, variables)
 
-        const query = `{
-          User( id: "u1" ) {
-            badgesCount
+        const query = gql`
+          {
+            User(id: "u1") {
+              badgesCount
+              badges {
+                key
+              }
+            }
           }
-        }
         `
-        const expected = { User: [{ badgesCount: 1 }] }
+        const expected = { User: [{ badgesCount: 1, badges: [{ key: 'indiegogo_en_rhino' }] }] }
 
         await expect(client.request(query)).resolves.toEqual(expected)
```


you don't need to be authenticated to query our user's then?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-11T19:51:43Z" title="Thursday, July 11th 2019, 9:51:43 pm +02:00">Jul 11, 2019</time>_

---


No, you don't need to be authenticated. Only when you request your own `email`.